appending advanced configuration to readme (#608)
1Password/shell-plugins: README.md

- Book a free [pairing session](https://calendly.com/d/grs-x2h-pmb/1password-shell-plugins-pairing-session) with one of our developers
- Join the [Developer Slack workspace](https://developer.1password.com/joinslack), and ask us any questions there

## Advanced Configuration
### Managing shell plugins in your own dotfiles

By default, `op plugin init` writes its configuration to `~/.config/op/plugins.sh`
and manages that file for you. If you'd rather keep your plugin configuration in
your own dotfiles (for example, tracked in version control), you can define the
shell functions yourself instead.

> ### 1. Define the plugin functions

Reviewer: ```suggestion

### Step 1: Define the plugin functions
```

To follow our typical style for formatting steps

> Shell functions for each shell should follow this pattern. In this case, 

Reviewer: Is there a specific action people need to take here? I'm not super familiar with this, but wondering if we could make it a bit more beginner friendly by giving an example of when/where to define these functions.

Author: This is a workaround for people who insist on maintaining their dotfiles using some management system, which usually requires storing the files in one place and sym-linking them to the directories applications (like 1Password) expect to find them. 

There are security reasons we are a bit careful with symlinked files so anyone taking this approach to their dotfiles has to jump through some hoops. This can mean manually defining the shell functions/aliases that wrap the CLI command (e.g., `gh`) in `op plugin run` since, if they're symlinking files, we're unable to write to the symlinked file. 

So this effectively documents manually performing what `op plugin init` would otherwise do automatically but can't in environments configured this way.

Author: Let me review the existing directions here and see if we can be a bit more hands-on with the guidance!

Author: I've made a minor change to clarify things a bit. 
TL;DR: as long as we are clear about what we expect, any user wanting to do this should _know_ how to do this according to their preference and the requirements of whatever they use to manage dotfiles. 

Happy to discuss this though! I might be a bit off the mark! 

Longer read: 
This is a rather power user need, and there's a lot of variation in exactly how people who want to centrally manage their dotfiles might do so. On our end, we just need to make sure users are clear on what 1Password expects (shell functions that follow the pattern documented, and are `sourced` by their shell config). Where they write those functions, how they source that file in their shell config, and where they store their shell config are largely user-defined, and anyone embarking on such an effort will likely have opinions about how to do this, or their dotfile manager will have opinions.

### bash / zsh

```bash
gh() {
  op plugin run -- gh "$@"
}
```

### fish

```fish
function gh --wraps gh --description "1Password shell plugin for GitHub CLI"
  op plugin run -- gh $argv
end
```

> ### 2. Tell `op` your plugins are already configured

Reviewer: ```suggestion
### Step 2: Tell 1Password CLI your plugins are already configured
```

To follow our typical style for how we refer to the CLI

> Set `OP_PLUGINS_SOURCED=1` so the 1Password CLI knows the shell plugins have

Reviewer: ```suggestion
Set `OP_PLUGINS_SOURCED=1` so 1Password CLI knows the shell plugins have
```
